### ISSUE TITLE
support l353

### ISSUE THREAD

**Issue by M1a2k3s4i5m6p** [mihai-dinculescu/tapo#266]
Хотелось бы иметь возможность менять цвет l353

**Comment by mihai-dinculescu** [comment]
Can you please confirm what lightbulb you're talking about? Google yielded no results for L353.
Do you mean L535?

**Comment by M1a2k3s4i5m6p** [comment]
yep l535*

**Comment by mihai-dinculescu** [comment]
Have you tried using the L530 API? Hopefully, they will be very similar or even identical.

**Comment by M1a2k3s4i5m6p** [comment]
yes i tried i can turn it on and off i can change the color temperature from warm to cold but i cant change the color for example to blue when i try to change the color nothing happens

**Comment by mihai-dinculescu** [comment]
That's interesting.
Can you please paste the output of `get_device_info_json`?

**Comment by M1a2k3s4i5m6p** [comment]
{'avatar': 'bulb', 'brightness': 100, 'color_temp': 3335, 'color_temp_range': [2500, 6500], 'default_states': {'re_power_type': 'always_on', 'state': {'brightness': 100, 'color_temp': 3335, 'hue': 0, 'saturation': 0}, 'type': 'custom'}, 'device_id': '802360D28BB87A639071AEB21D8DF42A22C7931F', 'device_on': True, 'dynamic_light_effect_enable': False, 'fw_id': '00000000000000000000000000000000', 'fw_ver': '1.1.5 Build 240328 Rel.194528', 'has_set_location_info': True, 'hue': 0, 'hw_id': '75A579FB67A2FFBEB6DB027CCBE46D1D', 'hw_ver': '3.0', 'ip': '192.168.0.25', 'lang': 'ru_RU', 'latitude': 571688, 'longitude': -21154, 'mac': '60-83-E7-4F-1A-4A', 'model': 'L535B', 'nickname': '0KPQvNC90LDRjyDQu9Cw0LzQv9Cw', 'oem_id': '0372A16EEBC5E511A0DF4915D6CCBCD9', 'overheated': False, 'region': 'Europe/London', 'rssi': -52, 'saturation': 0, 'signal_level': 2, 'specs': '', 'ssid': 'U0tZQ0hKUFE=', 'time_diff': 0, 'type': 'SMART.TAPOBULB'}

**Comment by avadiax** [comment]
Hi @mihai-dinculescu  I have the same issue. I used to have this command working well in 0.4.0.

await light.set().brightness(80).color(Color.DeepSkyBlue).send(light)

... and it will show blue colour.

Upgrade to 0.5.0 ... this command just turns on the light to white color. Running the example script "tapo_l530.py" shows that all colour display are not working correctly.

Looking at the changes in [tapo/src/requests/set_device_info/color_light.rs](https://github.com/mihai-dinculescu/tapo/commit/a61741ae11fcddb409bbf755f8a786d46cd0e7c3#diff-e6ee32fb24f54db591886ec7986810cdcab59fab82c1b225b3e9e4e50562aa30)

I don't think setting the below is correct. Temperature cannot be None ... it should be 0.

        self.color_temperature = None;

**Comment by mihai-dinculescu** [comment]
@avadiax What device do you have?
I've tested it with an L530 hw v1, and it works fine, but other coloured lightbulbs might behave slightly differently.

**Comment by avadiax** [comment]
> @avadiax What device do you have? I've tested it with an L530 hw v1, and it works fine, but other coloured lightbulbs might behave slightly differently.

I have L535 v3.0 firmware 1.1.5 b240328 Rel 194528.

**Comment by avadiax** [comment]
Hi @mihai-dinculescu ... I have new discoveries.

I downgraded the library to 0.4.0 to run the "l530.py" script across 4 bulbs ... they all work (to see if I have the same issue with other bulbs where authentication may fail like what you experienced).

Guess what ... I upgraded back to 0.5.0, tested the script again ... and it works for all bulbs ... but there is one caveat. It only works once. Once you re-run the script, it starts to show white display again.

So to test the issue.

1. Run "l530.py" script using 0.4.0. It should work normally.
2. Upgrade library to 0.5.0. Run the script again, it should work once.
3. Run the script one more time, it starts to show white light for all colours.
4. Downgrade back to 0.4.0 ... it works normally again.

I will look into the .off function to see it does anything different.

**Comment by mihai-dinculescu** [comment]
My login issue was magically fixed a few minutes later, and it never came back.
It was a new lightbulb with a newly set DHCP Address Reservation. Therefore, there are quite a few possible explanations, so I'm not going to dig further into it unless I hear about a similar issue.

The regression has been fixed on main.
